app/src/lib/markdown-filters/node-filter.ts

```diff
@@ -33,7 +40,11 @@ export interface INodeFilter {
 export function buildCustomMarkDownNodeFilterPipe(
   emoji: Map<string, string>
 ): ReadonlyArray<INodeFilter> {
-  return [new EmojiFilter(emoji)]
+  if (customMarkDownNodeFilterPipe === undefined) {
+    customMarkDownNodeFilterPipe = [new EmojiFilter(emoji)]
+  }
+
+  return customMarkDownNodeFilterPipe
 }
```

Comment:
I think we can get rid of the intermediate `customMarkdownFillterPipe` by leveraging `memoizeOne` here.

```ts
export const buildCustomMarkDownNodeFilterPipe = memoizeOne(
  (emoji: Map<string, string>): ReadonlyArray<INodeFilter> => [
    new EmojiFilter(emoji),
  ]
)
```

This will also ensure that we'll build a new pipe for each emoji map (by reference equality) should we happen to use multiple in the future.